sound/soc/sof/sof-audio.h

```diff
@@ -182,6 +205,7 @@ struct snd_sof_dai {
 	bool configured; /* DAI configured during BE hw_params */
 	struct sof_ipc_dai_config *dai_config;
 	struct list_head list;	/* list in sdev dai list */
+	void *private;
```

Comment:
ok, so something looks off. You already have a field line 206 for struct sof_ipc_dai_config *dai_config;

and now you're adding it again in the private data?

struct sof_dai_private_data {
	struct sof_ipc_comp_dai *comp_dai;
	struct sof_ipc_dai_config *dai_config;
};

Reply:
only temporarily. The ops for IPC3 parse tokens and save the data in comp_dai/dai_config that are part of the private_data but since the ops are not used in this patch, we retain the original comp_dai and dai_config in struct snd_sof_dai as well. They will be removed in the next patch. I updated the commit message to mention this